cloud_Azure/terraform/module/network_watcher.tf

```diff
@@ -1,32 +1,62 @@
-# Creates network watcher
-resource "azurerm_network_watcher" "kentik_network_watcher" {
-  name                = "${var.prefix}_network_watcher"
-  location            = var.location
-  resource_group_name = var.resource_group_name
+# Created automatically by Azure when VirtualNetwork is created or updated in the subscription, see:
+# https://docs.microsoft.com/en-us/azure/network-watcher/network-watcher-create
+data "azurerm_network_watcher" "network_watcher" {
+  name                = "NetworkWatcher_${var.location}"
+  resource_group_name = "NetworkWatcherRG"
 }
 
-# Runs python script to gather network security groups and expose it as a data source
+# Runs python script to gather network security groups from each requested resource group
```

Comment:
It is surprising for me that the Python script is needed at all.

Reply:
I suppose the terraform providers only expose a subset of CLI functionality and sadly they don't allow to list existing network security group.

Reply:
I'm fine with the current solution, However, I've found that [azurerm_resources data source](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/resources) allows access to various other aspects of an Azure deployment. The complete list of resource types is [here](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers). There is no resource named "network security group" or "NSG", but maybe we could get what we need via "Microsoft.Network Network Watcher" or "Microsoft.Network Virtual Network". I'm though absolutely not sure that there is any viable solution here. This is just FYI.